### ISSUE TITLE
Can we get rid of the "withHint" stuff?

### ISSUE THREAD

**Issue by chrishobcroft** [livepeer/protocol#594]
The `withHint` stuff results in overloading RPC endpoints with requests for the entire list of Orchestrators in order to look up the previous and subsequent Orchestrators in the ranking.

Since moving to Arbitrum, do we still need this temporary solution to "L1 gas fees too high"

cc. @kyriediculous

**Comment by 0xcadams** [comment]
This still results in gas savings for users - it is a net positive for users to pay less by moving computation off-chain, even on Arbitrum. This shouldn't be an issue since contract reads are free?

**Comment by chrishobcroft** [comment]
Right, reads are free in that they don't cost money, but they do cost something: if only in terms of power and bandwidth, plus querying some RPC endpoint which queries chainstate data somewhere.

Reads kinda do cost, in terms of user attention, responsivity, slick UX etc. In some ways your reputation is at stake, so you'd better be readable otherwise game over.

And this is the motivation behind this issue. The `withHint` stuff, for me, is technical debt, built up to solve for high inclusion fees, but now putting excess load on another part of this system, which has some kind of centralising force, with community convening around a "community Arbitrum" node, in order to maintain operatability.

But at some level, you're right. Both options are available as functions on the contract, with hint and without.
